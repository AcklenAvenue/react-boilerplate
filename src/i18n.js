const appLocales = ['en', 'es'];
const defaultLocale = 'es';

const messages = appLocales.reduce((languageMessages, language) => {
  // $FlowFixMe
  languageMessages[`${language}`] = require(`./locales/${language}`);
  return languageMessages;
}, {});

module.exports = {
  appLocales,
  defaultLocale,
  messages
};
