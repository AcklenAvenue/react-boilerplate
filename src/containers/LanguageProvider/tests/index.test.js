import React from 'react';
import renderer from 'react-test-renderer';

import { LanguageProvider, mapStateToProps } from '../index';

import { messages } from '../../../i18n';

describe('<LanguageProvider />', () => {
  it('should render its children', () => {
    const children = <h1>Test</h1>;
    const renderedComponent = renderer.create(
      <LanguageProvider messages={messages} locale="en">
        {children}
      </LanguageProvider>
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<LanguageProvider />', () => {
  it('Map state to props', () => {
    const state = {
      languageProvider: {
        locale: 'es',
        messages: {}
      }
    };
    const props = mapStateToProps(state);
    expect(props).toEqual({ locale: 'es' });
  });
});
