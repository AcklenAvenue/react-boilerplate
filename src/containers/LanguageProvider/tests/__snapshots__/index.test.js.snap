// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageProvider /> should render its children 1`] = `
<h1>
  Test
</h1>
`;
